src/custom/hooks/useSwapCallback.ts

```diff
@@ -163,6 +180,7 @@ export function useSwapCallback(
           addPendingOrder,
           signer: library.getSigner(),
         })
+        postOrderPromise.finally(closeModals)
```

Comment:
Because imo it's also easy to read (it's very declarative), less verbose than try/catch in this case, it also don't increment the cyclomatic complexity when can be easily avoidable. 

Also, cause try-catch would require me to:
- await for the promise (note we are not doing that)
- duplicate the `closeModals` (add it in the main try block, and in the catch. Here I do it once.